### PR TITLE
Chain pump fixed

### PR DESCRIPTION
**Pull request by YodaMike**

### CODE REVIEW

File: search-api/src/main/kotlin/fund/cyber/search/configuration/AppContext.kt [hunk 1]
```diff
@@ -1,18 +1,24 @@
...
 import org.elasticsearch.common.transport.InetSocketTransportAddress
 import org.elasticsearch.transport.client.PreBuiltTransportClient
 import java.net.InetAddress
 
 object AppContext {
 
-    val jsonSerializer = ObjectMapper()
-    val jsonDeserializer = ObjectMapper()
+    val jsonSerializer = ObjectMapper().registerKotlinModule()
+            .registerModule(Jdk8Module())
```
Comment (vstavetski) [inline search-api/src/main/kotlin/fund/cyber/search/configuration/AppContext.kt]: use own ObjectMapper for each hendler

File: pumps/src/main/kotlin/fund/cyber/pump/BlockchainEventsGenerator.kt [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+import fund.cyber.node.common.StackCache
+
+fun Flowable<BlockchainInterface<*>>.generateEvents(startBlockNumber: Long): Flowable<out BlockchainEvent> {
+    return this.flatMap {blockchain ->
+        fund.cyber.pump.generateEvents(blockchain, startBlockNumber)
+    }
+}
+
+fun generateEvents(blockchain: BlockchainInterface<*>, startBlockNumber: Long): Flowable<out BlockchainEvent> {
+    val fl = ConcurrentPulledBlockchain(blockchain)
```
[the remark this answers is not in the document]
Comment (YodaMike) [reply pumps/src/main/kotlin/fund/cyber/pump/BlockchainEventsGenerator.kt]: done

File: pumps/src/main/kotlin/fund/cyber/pump/BlockchainEventsGenerator.kt [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    val context = ConvertContext(
+            exHash = "",
+            history = StackCache(20),
```
[the remark this answers is not in the document]
Comment (YodaMike) [reply pumps/src/main/kotlin/fund/cyber/pump/BlockchainEventsGenerator.kt]: done

File: pumps/src/main/kotlin/fund/cyber/pump/BlockchainEventsGenerator.kt [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    )
+
+    return fl.subscribeBlocks(startBlockNumber)
+            .flatMap { blockBundle ->
+                return@flatMap convert(blockBundle, context).toFlowable()
+            }
+}
+
+fun convert(_blockBundle: BlockBundle, context: ConvertContext): List<BlockchainEvent> {
+    var blockBundle = _blockBundle
```
[the remark this answers is not in the document]
Comment (YodaMike) [reply pumps/src/main/kotlin/fund/cyber/pump/BlockchainEventsGenerator.kt]: done

File: pumps/src/main/kotlin/fund/cyber/pump/BlockchainEventsGenerator.kt [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    var events: List<out BlockchainEvent>
+
+    if (context.exHash == "" || context.exHash == blockBundle.parentHash) {
+        context.exHash = blockBundle.hash
+        context.history.push(blockBundle)
+        events = listOf(CommitBlock(blockBundle))
+    } else {
+        var exBl: BlockBundle? = null
```
[the remark this answers is not in the document]
Comment (YodaMike) [reply pumps/src/main/kotlin/fund/cyber/pump/BlockchainEventsGenerator.kt]: done

File: search-api/src/main/kotlin/fund/cyber/search/configuration/AppContext.kt (hunk 1 above)
Comment (YodaMike) [reply search-api/src/main/kotlin/fund/cyber/search/configuration/AppContext.kt]: done

**Pull request merged**